predicators/pretrained_model_interface.py

```diff
@@ -5,16 +5,20 @@
 """
 
 import abc
+import base64
 import logging
 import os
 import time
+from io import BytesIO
 from typing import List, Optional
 
+import cv2
 import google
 import google.generativeai as genai
 import imagehash
 import openai
 import PIL.Image
+from tenacity import retry, stop_after_attempt, wait_random_exponential
```

Comment:
this needs to be added as a requirement in setup.py and potentially some other places

Reply:
will do